hack/cluster-sync.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2021 Red Hat, Inc.
+#
+
+set -e
+
+TEMP_FILE=$(mktemp -p /tmp -t kubevirt.deploy.XXXX)
+
+trap 'rm -f $TEMP_FILE' EXIT SIGINT
+
+function main() {
+    ./hack/cluster-clean.sh --skip-wait >$TEMP_FILE 2>&1 &
+    CLEAN_PID=$!
+
+    make cluster-build
+
+    echo "waiting for cluster-clean to finish"
+    if ! wait $CLEAN_PID; then
+        echo "cluster-clean failed, output was:"
+        cat $TEMP_FILE
+        exit 1
+    fi
+
+    ./hack/cluster-clean.sh --skip-clean
```

Comment:
Calling cluster-clean with a "skip-clean" flag, that is very odd and confusing.
You should consider sourcing that script and calling specific function instead of using these flags (leaving `main` as is). Considering that the usage of this is not likely to be "public", it fits well.

Reply:
I prefer to keep it this way if possible, and change vars to `--clean` , `--wait`

Reply:
The comment gave reasoning on why this needs to be reconsidered. What is the reasoning to keep it?

Reply:
I think its simpler like this, and self contained (all is encapsulated in one place, and doesnt break the previous calling method)
we will have one script that can be run as a whole with the current `make clean` from the makefile itself without need to source it and then run it (which is less clean in makefiles)

Reply:
This is why I said not to touch `main` logic (just to call those two functions without a condition), and to create two other functions and call them directly. The current users will not be affected.

Having one function, that is split to two in practice and adding an argument parser is not nice when an alternative is in hand to do it better.
They main problem is the look of this from the caller side, calling a clean with a "skip-clean" flag makes no sense to me.

Reply:
first let me please update he variables and then we reconsider

2. if we source it when it has main then it will run the main function right ? this what i meant is the problem,
unless we do leave the main and preset control variables before sourcing it

Reply:
Right, I see now. Sourcing as is will not work well.

A "no-wait" flag is fine (`only-clean` I do not like), but the one that skips the cleaning is too odd to "eat".
Maybe move the "wait" code to one of the other sources scripts (or a new one), and then you can call it.

Reply:
> A "no-wait" flag is fine (`only-clean` I do not like), but the one that skips the cleaning is too odd to "eat".
> Maybe move the "wait" code to one of the other sources scripts (or a new one), and then you can call it.

I prefer to not over engineer it,  and do keep it in one file
what about this ? (and then we can back to your original idea)
```
# in the sourcer
. ./script.sh --source-only

# in the script itself
if [ "${1}" != "--source-only" ]; then
    main "${@}"
fi
```